### DISCUSSION TITLE
List of classification methods

### DISCUSSION THREAD

**Discussion by Boblancvich** [online-ml/river#1111] [Q&A] [answered]
Hi,
Where would I be able to find a full list of Classification methods that River covers?

Thank you!

**Comment by MaxHalford** [comment, marked as answer]
Hey @Boblancvich!

Alas we don't have a nice list in our docs. But you can do this:

```py
import importlib
import inspect
from river import base

for submodule in importlib.import_module("river.api").__all__:

    def is_classifier(obj):
        return inspect.isclass(obj) and issubclass(obj, base.Classifier)

    for name, obj in inspect.getmembers(
        importlib.import_module(f"river.{submodule}"), is_classifier
    ):
        print(f"{submodule}.{name}")
```

```
base.Classifier
base.MiniBatchClassifier
compat.SKL2RiverClassifier
dummy.NoChangeClassifier
dummy.PriorClassifier
ensemble.ADWINBaggingClassifier
ensemble.AdaBoostClassifier
ensemble.AdaptiveRandomForestClassifier
ensemble.BaggingClassifier
ensemble.LeveragingBaggingClassifier
ensemble.SRPClassifier
ensemble.StackingClassifier
ensemble.VotingClassifier
facto.FFMClassifier
facto.FMClassifier
facto.FwFMClassifier
facto.HOFMClassifier
imblearn.HardSamplingClassifier
imblearn.RandomOverSampler
imblearn.RandomSampler
imblearn.RandomUnderSampler
linear_model.ALMAClassifier
linear_model.LogisticRegression
linear_model.PAClassifier
linear_model.Perceptron
linear_model.SoftmaxRegression
model_selection.SuccessiveHalvingClassifier
multiclass.OneVsOneClassifier
multiclass.OneVsRestClassifier
multiclass.OutputCodeClassifier
multioutput.ClassifierChain
multioutput.MonteCarloClassifierChain
multioutput.ProbabilisticClassifierChain
naive_bayes.BernoulliNB
naive_bayes.ComplementNB
naive_bayes.GaussianNB
naive_bayes.MultinomialNB
neighbors.KNNClassifier
tree.ExtremelyFastDecisionTreeClassifier
tree.HoeffdingAdaptiveTreeClassifier
tree.HoeffdingTreeClassifier
tree.LabelCombinationHoeffdingTreeClassifier
tree.SGTClassifier
```

I hope that helps!

**Reply by Boblancvich** [reply]
Hi Max,
Thanks a lot for this!

As there are quite a few, which would you recommend are the best for working with text processing?

Thanks!

**Reply by MaxHalford** [reply]
I would recommend first trying the models in `naive_bayes`, as well as `LogisticRegression`. You'll want to use the `TFIDF` in the `feature_extraction` module too :)

**Reply by Dennis1989** [reply]
Agreed. Both have a strong baseline performance and are rarely beaten by anything other than language models. 

Maybe it would also make sense to transform the input-sentences into average word embeddings and then use logreg or Naive Bayes on these representations.

**Reply by MaxHalford** [reply]
@raphaelsty IIRC you had an example of using word embeddings with River? I can't seem to find in the docs 😅

**Reply by smastelini** [reply]
Hey @Dennis1989, in your experience are sentence BERT (SBERT) models also a good idea to vectorize text data? Maybe they are too slow for streaming processing

**Reply by Dennis1989** [reply]
@smastelini In general, yes, it is a good idea! SBERT especially performs super well, at least for traditional clustering. For stream clustering, I never considered them as inference is slow. Additionally, language models are not really updated (also word embeddings), which is quite problematic in our domain because of concept drift. I played around with incremental embeddings, which are promising but still WIP...

**Reply by Boblancvich** [reply]
Great, thanks for the advice guys :)

**Reply by Boblancvich** [reply]
Hi again,

I was also curious as to why there is no Online version of SVM? 

Thanks!

**Reply by MaxHalford** [reply]
Good question. Actually, there is. If you use the logistic regression with a Hinge loss, that results in a linear SVM. Then, if you need to handle non-linearity, then it's recommended to use the `RBFSampler` class from the `feature_extraction` module :)

**Reply by Boblancvich** [reply]
Hi @MaxHalford ,

Thanks for this idea! I'll give it a go!
